plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessor.java

```diff
@@ -1988,14 +1988,20 @@ public Answer copyVolumeFromPrimaryToPrimary(CopyCommand cmd) {
             }
 
             destPool = storagePoolMgr.getStoragePool(destPrimaryStore.getPoolType(), destPrimaryStore.getUuid());
-            storagePoolMgr.copyPhysicalDisk(volume, destVolumeName, destPool, cmd.getWaitInMillSeconds());
-
-            if (srcPrimaryStore.isManaged()) {
-                storagePoolMgr.disconnectPhysicalDisk(srcPrimaryStore.getPoolType(), srcPrimaryStore.getUuid(), srcVolumePath);
-            }
+            try {
+                storagePoolMgr.copyPhysicalDisk(volume, destVolumeName, destPool, cmd.getWaitInMillSeconds());
+            } catch (Exception e) {
+                String errMsg = "Failed to copy volume: " + srcVol.getName() + " to dest storage: " +  destPrimaryStore.getName() + ", due to " + e.toString();
+                s_logger.error(errMsg, e);
```

Comment:
only log the stack trace in debug?
```suggestion
                s_logger.error(errMsg);
                s_logger.debug(errMsg, e);
```